src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioTaskSchedulerFactory.cs

```diff
@@ -94,7 +94,7 @@ protected override void QueueTask(Task task)
             {
                 _joinableTaskFactory.RunAsync(async () =>
                 {
-                    await _joinableTaskFactory.SwitchToMainThreadAsync();
+                    await _joinableTaskFactory.SwitchToMainThreadAsync(alwaysYield: true);
```

Comment:
not trying to pick anything. just try to understand. so, this means in VS, VSWorkspace events will fire one by one with yield in it? with Jason's batch might be okay, but in solution load before, how does it work? won;t it take really long time to fire all those events?